internal/confmapprovider/discovery/properties/conf_test.go

```diff
@@ -0,0 +1,115 @@
+	for _, path := range []string{
+		filepath.Join(".", "testdata", "valid-mapping.yaml"),
+		filepath.Join(".", "testdata", "valid-set.yaml"),
+		filepath.Join(".", "testdata", "valid-mix.yaml"),
+	} {
+		t.Run(filepath.Base(path), func(t *testing.T) {
+			conf, err := confmaptest.LoadConf(path)
+			require.NoError(t, err)
+			loaded, warning, fatal := LoadConf(conf.ToStringMap())
+			require.NoError(t, warning)
+			require.NoError(t, fatal)
+			require.Equal(t, map[string]any{
+				"extensions": map[string]any{
+					"docker_observer": map[string]any{
+						"enabled": "false",
+					},
+					"host_observer/with_a_name": map[string]any{
+						"refresh_interval": "1h",
+					},
+				},
+				"receivers": map[string]any{
+					"a_receiver": map[string]any{
+						"config": map[string]any{
+							"some_field": "some_value",
+						},
+					},
+					"another_receiver/with-name": map[string]any{
+						"config": map[string]any{
+							"parent": map[string]any{
+								"child_one": map[string]any{
+									"another_field": "another_value",
+									"child_two": map[string]any{
+										"another_field":     "another_value",
+										"yet_another_field": "yet_another_value",
+									},
+								},
+							},
+						},
+					},
+				},
+			}, loaded.ToStringMap())
+		})
+	}
+}
+
+func TestInvalidConf(t *testing.T) {
+	for _, fixture := range []struct {
+		expectedError string
+		path          string
+	}{
+		{
+			expectedError: `unknown property "splunk.discovery.receivers.a_receiver.unknown"; unknown property "splunk.discovery.receivers.another_receiver/with-name.another_unknown"`,
+			path:          filepath.Join(".", "testdata", "invalid-mapping-unknown-subfield.yaml"),
+		},
+		{
+			expectedError: `unknown discovery property mapping "splunk.discovery.123"; unknown discovery property mapping "splunk.discovery.processors"`,
+			path:          filepath.Join(".", "testdata", "invalid-mapping-unknown-component.yaml"),
+		},
+		{
+			expectedError: `invalid property "splunk.discovery.123" (parsing error): splunk.discovery:1:18: unexpected token "123" (expected ("receivers" | "extensions") <dot> ComponentID <dot> (("config" <dot>) | "enabled") (<string> | <dot> | <forwardslash>)*); invalid property "splunk.discovery.processors.a_processor.config.some_field" (parsing error): splunk.discovery:1:18: unexpected token "processors" (expected ("receivers" | "extensions") <dot> ComponentID <dot> (("config" <dot>) | "enabled") (<string> | <dot> | <forwardslash>)*); invalid property "splunk.discovery.receivers.a_receiver.unknown.some_field" (parsing error): splunk.discovery:1:57: unexpected token "<EOF>" (expected <dot> (("config" <dot>) | "enabled") (<string> | <dot> | <forwardslash>)*); invalid property "splunk.discovery.receivers.another_receiver/with-name.another_unknown.some_other_field" (parsing error): splunk.discovery:1:87: unexpected token "<EOF>" (expected <dot> (("config" <dot>) | "enabled") (<string> | <dot> | <forwardslash>)*)`,
```

Comment:
Maybe add a comment about how this test expects to work.

Reply:
renamed the test to be more descriptive